crates/era-cheatcodes/src/cheatcodes.rs

```diff
@@ -2578,6 +2705,243 @@ impl CheatcodeTracer {
         );
         modified_bytecodes
     }
+
+    /// Initializes the EVM fork
+    /// Translates all deployed contract bytecodes with EVM bytecodes
+    /// Sets the EVM env to use for execution with the latest fork environment
+    fn maybe_switch_vm<S: DatabaseExt + DatabaseCommit + Send>(
+        &mut self,
+        mut db: MutexGuard<'_, Box<S>>,
+        fork_id: LocalForkId,
+        force_initialized: bool,
+    ) {
```

Comment:
I'd love to change the target of this function and name it switch_to_evm and check everything outside. 

It's always difficut to understand what is going on, when you check one main condition at the top of the function and do almost the whole logic under the bracket. 

Please consider separate this functionality

Reply:
Replaced `maybe_switch_vm` with `select_fork_vm` as this better conveys the relationship between selecting a fork (`select_fork`) and immediately selecting a vm for it.